File: sdk/csharp/examples/Blockchain.cs
// Blockchain operations with the HyperMesh C# SDK.
//
// Run: dotnet run

using HyperMesh.Sdk;

using var client = new HyperMeshClient();

// Get chain height
var h = await client.Blockchain.HeightAsync();
Console.WriteLine($"Blockchain height: {h.Height}");

// Fetch the genesis block
var genesis = await client.Blockchain.BlockAsync(0);
Console.WriteLine($"\nGenesis block:");
Console.WriteLine($"  Index: {genesis.Index}");
Console.WriteLine($"  Hash: {genesis.Hash}");
Console.WriteLine($"  Previous hash: {genesis.PreviousHash}");
Console.WriteLine($"  Timestamp: {genesis.Timestamp}");

// Fetch the latest block
if (h.Height > 0)
{
    var latest = await client.Blockchain.BlockAsync(h.Height - 1);
    Console.WriteLine($"\nLatest block (index {latest.Index}):");
    Console.WriteLine($"  Hash: {latest.Hash}");
}

// Validate the chain
var result = await client.Blockchain.ValidateAsync();
Console.WriteLine($"\nBlockchain valid: {result.Valid}");
Console.WriteLine($"Blocks checked: {result.BlockCount}");
if (result.Errors.Count > 0)
{
    Console.WriteLine($"Errors: {string.Join(", ", result.Errors)}");
}

File: sdk/python/examples/blockchain.py
"""Blockchain operations with the HyperMesh Python SDK.

Run: python examples/blockchain.py
"""

from hypermesh import HyperMeshClient


def main() -> None:
    client = HyperMeshClient()

    # Get chain height
    h = client.blockchain.height()
    print(f"Blockchain height: {h.height}")

    # Fetch the genesis block
    genesis = client.blockchain.block(0)
    print(f"\nGenesis block:")
    print(f"  Index: {genesis.index}")
    print(f"  Hash: {genesis.hash}")
    print(f"  Previous hash: {genesis.previous_hash}")
    print(f"  Timestamp: {genesis.timestamp}")
    print(f"  Data: {genesis.data}")

    # Fetch the latest block
    if h.height > 0:
        latest = client.blockchain.block(h.height - 1)
        print(f"\nLatest block (index {latest.index}):")
        print(f"  Hash: {latest.hash}")

    # Validate the chain
    result = client.blockchain.validate()
    print(f"\nBlockchain valid: {result.valid}")
    if result.errors:
        print(f"Errors: {result.errors}")


if __name__ == "__main__":
    main()
